src/wp-includes/rest-api/endpoints/class-wp-rest-font-families-controller.php

```diff
@@ -0,0 +1,280 @@
+<?php
+/**
+ * Rest Font Families Controller.
+ *
+ * This file contains the class for the REST API Font Families Controller.
+ *
+ * @package    WordPress
+ * @subpackage Font Library
+ * @since      6.4.0
+ */
+
+/**
+ * Font Families Controller class.
+ *
+ * @since 6.4.0
+ */
+class WP_REST_Font_Families_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 6.4.0
+	 */
+	public function __construct() {
+		$this->rest_base = 'font-families';
+		$this->namespace = 'wp/v2';
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @since 6.4.0
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'install_font' ),
+					'permission_callback' => array( $this, 'update_font_library_permissions_check' ),
+					'args'                => array (
+						'slug'       => array(
+							'required' => true,
+							'type' => 'string',
+						),
+						'name'       => array(
+							'required' => true,
+							'type' => 'string',
+						),
+						'fontFamily' => array(
+							'required' => true,
+							'type' => 'string',
+						),
+						'fontFace'   => array(
+							'required' => false,
+							'type' => 'string',
+							'validate_callback' => array( $this, 'validate_font_faces' ),
+						),
+					)
+				),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\/\w-]+)',
+			array(
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'uninstall_font' ),
+					'permission_callback' => array( $this, 'update_font_library_permissions_check' ),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Returns validation errors in font faces data for installation.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array[] $font_faces Font faces to install.
+	 * @param array   $files         Files to install.
+	 * @return WP_Error Validation errors.
+	 */
+	private function get_validation_errors( $font_faces, $files ) {
+		$error = new WP_Error();
+
+		if ( ! is_array( $font_faces ) ) {
+			$error->add( 'rest_invalid_param', __( 'fontFace should be an array.' ) );
+			return $error;
+		}
+
+		if ( count( $font_faces ) < 1 ) {
+			$error->add( 'rest_invalid_param', __( 'fontFace should have at least one item.' ) );
+			return $error;
+		}
+
+		for ( $face_index = 0; $face_index < count( $font_faces ); $face_index++ ) {
+			$font_face = $font_faces[ $face_index ];
+			if ( ! isset( $font_face['fontWeight'] ) || ! isset( $font_face['fontStyle'] ) ) {
+				$error_message = sprintf(
+					// translators: 1: font face index.
+					__( 'Font face (%1$s) should have fontWeight and fontStyle properties defined.' ),
```

Comment:
Use placeholder for `fontWeight` and `fontStyle` as well. Use numbered placeholders. Update translator comment.